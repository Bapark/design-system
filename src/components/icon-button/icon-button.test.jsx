import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import IconButton from './icon-button.jsx';

describe('<IconButton />', () => {
  const requiredProps = {
    label: 'Test label',
    icon: {
      id: 'test-id',
      viewBox: '1 2 3 4',
    },
    onClick: () => {},
  };

  it('has defaults', () => {
    render(<IconButton {...requiredProps} />);
    expect(document.body).toMatchSnapshot();
  });

  describe('label prop API', () => {
    it('can be set', () => {
      render(<IconButton {...requiredProps} label="Unique label" />);
      expect(screen.getByRole('button', { name: 'Unique label' })).toBeInTheDocument();
    });
  });

  describe('icon prop API', () => {
    it('sets the xlink:href', () => {
      render(<IconButton {...requiredProps} icon={{ ...requiredProps.icon, id: 'unique-id' }} />);
      expect(screen.getByRole('button', { name: 'Test label' }).children[1]).toHaveAttribute('xlink:href', '#unique-id');
    });

    it('sets the width', () => {
      render(<IconButton {...requiredProps} icon={{ ...requiredProps.icon, viewBox: '1 2 3 41'}} />);
      expect(screen.getByRole('button', { name: 'Test label' })).toHaveAttribute('height', '41');
    });

    it('sets the width', () => {
      render(<IconButton {...requiredProps} icon={{ ...requiredProps.icon, viewBox: '1 2 42 4'}} />);
      expect(screen.getByRole('button', { name: 'Test label' })).toHaveAttribute('width', '42');
    });
  });

  describe('onClick prop API', () => {
    it('can be set', () => {
      const onClickSpy = jest.fn();
      render(<IconButton {...requiredProps} onClick={onClickSpy} />);
      userEvent.click(screen.getByRole('button', { name: 'Test label' }));
      expect(onClickSpy).toHaveBeenCalledWith(expect.anything());
    });
  });
});
