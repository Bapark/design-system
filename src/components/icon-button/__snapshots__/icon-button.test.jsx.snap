// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> has defaults 1`] = `
<body>
  <div>
    <svg
      class="icon-base"
      height="4"
      role="button"
      tabindex="0"
      width="3"
    >
      <title>
        Test label
      </title>
      <use
        xlink:href="#test-id"
      />
    </svg>
  </div>
</body>
`;
